lib/mix/test/mix/tasks/format_test.exs

```diff
@@ -174,6 +184,61 @@ defmodule Mix.Tasks.FormatTest do
     end
   end
 
+  test "can read exported configuration from dependencies", context do
+    Mix.Project.push(__MODULE__.FormatWithDepsApp)
+
+    in_tmp context.test, fn ->
+      File.write!(".formatter.exs", """
+      [import_deps: [:my_dep]]
+      """)
+
+      File.write!("a.ex", """
+      my_fun :foo, :bar
+      """)
+
+      File.mkdir_p!("deps/my_dep/")
+
+      File.write!("deps/my_dep/.formatter.exs", """
+      [export: [locals_without_parens: [my_fun: 2]]]
+      """)
+
+      Mix.Tasks.Format.run(["a.ex"])
+
+      assert File.read!("a.ex") == """
+             my_fun :foo, :bar
+             """
+
+      manifest_path = Path.join(Mix.Project.manifest_path(), ".cached_deps_formatter")
+      assert File.regular?(manifest_path)
+
+      # Let's check that the manifest gets updated if .formatter.exs changes.
+      File.touch!(manifest_path, {{1970, 1, 1}, {0, 0, 0}})
+      %File.Stat{mtime: mtime} = File.stat!(manifest_path)
+
+      File.touch!(".formatter.exs")
+      Mix.Tasks.Format.run(["a.ex"])
+      %File.Stat{mtime: new_mtime} = File.stat!(manifest_path)
+
+      assert new_mtime > mtime
```

Comment:
Other places in the suite we would do:

    assert File.stat!(manifest_path).mtime > {{1970, 1, 1}, {0, 0, 0}}